gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala

```diff
@@ -276,6 +276,13 @@ case class TransformPreOverrides(isAdaptiveContext: Boolean)
             } else {
               return shj.withNewChildren(shj.children.map(replaceWithTransformerPlan))
             }
+          case scan: FileSourceScanExec =>
+            val newPartitionFilters = {
+              ExpressionConverter.transformDynamicPruningExpr(scan.partitionFilters, reuseSubquery)
+            }
+            val newSource = scan.copy(partitionFilters = newPartitionFilters)
+            TransformHints.tagNotTransformable(newSource, "fallback scan")
```

Comment:
It looks `BatchScan/HiveTableScan` have the similar issue. It may be better to have a separate impl. like `applyScanTransformer` to handle these operators. The difference is scan will not be transformed. Thanks!

Reply:
@PHILO-HE done.